Argo/Functions/sequence.swift

```diff
@@ -1,11 +1,30 @@
 public func sequence<T>(xs: [Decoded<T>]) -> Decoded<[T]> {
-  return xs.reduce(pure([])) { accum, elem in
-    curry(+) <^> accum <*> ({ [$0] } <^> elem)
+  var accum: [T] = []
+  accum.reserveCapacity(xs.count)
+  
+  for elem in xs {
+    switch elem {
+    case let .Success(value):
+      accum.append(value)
+    case let .Failure(error):
+      return .Failure(error)
+    }
   }
+  
+  return pure(accum)
 }
 
 public func sequence<T>(xs: [String: Decoded<T>]) -> Decoded<[String: T]> {
-  return xs.reduce(pure([:])) { accum, elem in
-    curry(+) <^> accum <*> ({ [elem.0: $0] } <^> elem.1)
+  var accum: [String: T] = [:]
+  
+  for (key, value) in xs {
+    switch value {
+    case let .Success(unwrapped):
+      accum[key] = unwrapped
+    case let .Failure(error):
+      return .Failure(error)
+    }
   }
+  
```

Comment:
Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)